packages/wpcom-proxy-request/src/index.js

```diff
@@ -351,8 +351,8 @@ function onload() {
 function onmessage( e ) {
 	debug( 'onmessage' );
 
-	// safeguard...
-	if ( e.origin !== proxyOrigin ) {
+	// Filter out messages from iframes not equal to the iframe variable, different origins
+	if ( e.origin !== proxyOrigin || e.source !== iframe.contentWindow ) {
 		debug( 'ignoring message... %o !== %o', e.origin, proxyOrigin );
```

Comment:
This debug message still assumes that the only reason to ignore are mismatched origins. Can you log a different message if the windows don't match?

Reply:
Done!

Reply:
Thanks! The checks look much better now 🙂